packages/devtools-client-adapters/src/firefox/types.js

```diff
@@ -0,0 +1,285 @@
+ *
+ * @memberof firefox
+ * @static
+ */
+export type ActualLocation = {
+  source: SourcePayload,
+  line: number,
+  column?: number
+};
+
+/**
+ * Response from the `getFrames` function call
+ * @memberof firefox
+ * @static
+ */
+export type FramesResponse = {
+  frames: FramePacket[],
+  from: ActorId
+}
+
+/**
+ * Actions
+ * @memberof firefox
+ * @static
+ */
+export type Actions = {
+  paused: (Pause) => void,
+  resumed: (ResumedPacket) => void,
+  newSource: (Source) => void,
+  fetchEventListeners: () => void
+};
+
+/**
+ * Tab Target gives access to the browser tabs
+ * @memberof firefox
+ * @static
+ */
+export type TabTarget = {
+  activeConsole: {
+    evaluateJS: (Script, Function, ?{ frameActor?: FrameId }) => void
+  },
+  form: { consoleActor: any },
+  activeTab: {
+    navigateTo: (string) => Promise<*>,
+    reload: () => Promise<*>
+  }
+}
+
+/**
+ * Clients for accessing the Firefox debug server and browser
+ * @memberof firefox/clients
+ * @static
+ */
+
+/**
+ * DebuggerClient
+ * @memberof firefox
+ * @static
+ */
+export type DebuggerClient = {
+  _activeRequests: {
+    get: (any) => any,
+    delete: (any) => void
+  }
+}
+
+/**
+ * A grip is a JSON value that refers to a specific JavaScript value in the
+ * debuggee. Grips appear anywhere an arbitrary value from the debuggee needs
+ * to be conveyed to the client: stack frames, object property lists, lexical
+ * environments, paused packets, and so on.
+ *
+ * For mutable values like objects and arrays, grips do not merely convey the
+ * value's current state to the client. They also act as references to the
+ * original value, by including an actor to which the client can send messages
+ * to modify the value in the debuggee.
+ *
+ * @see https://wiki.mozilla.org/Remote_Debugging_Protocol#Grips
+ * @memberof firefox
+ * @static
+ */
+// FIXME: need Grip definition
+export type Grip = {
+
+}
+
+/**
+ * SourceClient
+ * @memberof firefox
+ * @static
+ */
+export type SourceClient = {
+  source: () => Source,
+  setBreakpoint: ({
+    line: number,
+    column?: number,
+    condition: boolean,
+    noSliding: boolean
+  }) => Promise<BreakpointResponse>,
+  prettyPrint: (number) => Promise<*>,
+  disablePrettyPrint: () => Promise<*>
+};
+
+/**
+ * ObjectClient
+ * @memberof firefox
+ * @static
+ */
+export type ObjectClient = {
+  getPrototypeAndProperties: () => any
+};
+
+/**
+ * ThreadClient
+ * @memberof firefox
+ * @static
+ */
+export type ThreadClient = {
+  resume: (Function) => Promise<*>,
+  stepIn: (Function) => Promise<*>,
+  stepOver: (Function) => Promise<*>,
+  stepOut: (Function) => Promise<*>,
+  breakOnNext: () => Promise<*>,
+  // FIXME: unclear if SourceId or ActorId here
+  source: ({ actor: SourceId }) => SourceClient,
+  pauseGrip: (Grip|Function) => ObjectClient,
+  pauseOnExceptions: (boolean, boolean) => Promise<*>,
+  interrupt: () => Promise<*>,
+  eventListeners: () => Promise<*>,
+  getFrames: (number, number) => FramesResponse
```

Comment:
In general I tried to stick to a `$XPacket` for information coming in as a notification or in response to a request.  And then `$XResponse` for the return value of a function call.